### PR TITLE
Koding DCOS Package

### PR DESCRIPTION
**Pull request by sibblegp**
A Koding (www.koding.com) Mesosphere DCOS Universe package.

Signed-off-by: George Sibble george@koding.com

### CODE REVIEW

File: repo/packages/K/koding/0/package.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "packagingVersion": "2.0",
```
Comment (ryadav88) [inline repo/packages/K/koding/0/package.json]: please use version 3.0

Comment (ryadav88) [inline repo/packages/K/koding/0/package.json]: Also add `minDcosReleaseVersion`

File: repo/packages/K/koding/0/package.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  "name": "koding",
+  "version": "0.0.1",
```
Comment (ryadav88) [inline repo/packages/K/koding/0/package.json]: `version` should be "kodingVersion-universePackageVersion". Ex: if koding's version for singular is 1.2.3 then `version` should be "1.2.3-0.0.1"

File: repo/packages/K/koding/0/package.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  "tags": ["mesosphere", "koding"],
```
Comment (ryadav88) [inline repo/packages/K/koding/0/package.json]: remove the mesosphere tag. It is reserved for packages created by Mesosphere

Comment (ryadav88) [inline repo/packages/K/koding/0/package.json]: you can use others like IDE or whatever keyword a user should use to find Koding's package

File: repo/packages/K/koding/0/package.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  "maintainer": "support@koding.com",
+  "description": "Koding is a web based collaborative software development tool.  We bring code development off of localhost and into the cloud.",
+  "scm": "https://github.com/koding/koding.git",
+  "website": "http://www.koding.com",
+  "framework": false,
+  "postInstallNotes": "Happy Koding!"
```
Comment (ryadav88) [inline repo/packages/K/koding/0/package.json]: Also add `preInstallNotes` and `postUninstallNotes`. 
For `preInstallNotes`: Add this line first - "This DC/OS Service is currently EXPERIMENTAL. There may be bugs, incomplete features, incorrect documentation, or other discrepancies.". 
Refer: https://github.com/mesosphere/universe/blob/version-3.x/repo/packages/O/openldap/3/package.json

If marathon-lb is needed for this package for work then highlight that too.

File: repo/packages/K/koding/0/package.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+}
```
Comment (ryadav88) [inline repo/packages/K/koding/0/package.json]: Add license info

File: repo/packages/K/koding/0/config.json [hunk 6]
```diff
@@ -0,0 +1,35 @@
+{
+  "type": "object",
+  "properties": {
+    "networking": {
```
Comment (ryadav88) [inline repo/packages/K/koding/0/config.json]: add name (name of the service instance), cpu (CPU share to allocate) and memory (memory to allocate) properties. Refer other packaged for this.

File: repo/packages/K/koding/0/config.json [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+      "description": "Configuration options for Koding's networking",
+      "type": "object",
+      "properties": {
+        "node": {
+          "description": "Which node to install Koding on",
```
Comment (ryadav88) [inline repo/packages/K/koding/0/config.json]: why do you need `node`?

**Comment by sibblegp** [comment]
@ryadav88 All changes made as requested except for Node.  See comment above.

**Comment by sibblegp** [comment]
Since a node is necessary (which I have seen in other packages), it won't be able to install with just the defaults but that is the only field which definitely needs to be filled in.

File: repo/packages/K/koding/0/config.json [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+          "description": "Whether Marathon-lb should redirect HTTP traffic to HTTPS. This requires 'virtual-host' to be set. By default, this is false.",
+          "type": "boolean",
+          "default": false
+        }
+      },
+      "required": [
+        "node",
+        "hostname",
+        "virtual-host",
+        "https-redirect"
```
Comment (ryadav88) [inline repo/packages/K/koding/0/config.json]: `https-redirect` shouldn't be required

**Comment by sibblegp** [comment]
@ssk2 Any update on when you'll be able to take a look and approve this?

**Comment by ryadav88** [comment]
@sibblegp Can you also add a link to the example at the end of `description` property in `package.json`
```Installation documentation: https://github.com/dcos/examples/tree/master/1.8/koding```
Ready to merge after that.

**Comment by sibblegp** [comment]
@ryadav88 Link added.

**Pull request merged**